core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -54,3 +58,40 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     Configuration.setConfiguration(null)
   }
 }
+
+object ZooKeeperTestHarness {
+
+  /**
+   * Verify that a previous test that doesn't use ZooKeeperTestHarness hasn't left behind an unexpected thread
+   */
+  @BeforeClass
+  def setUpClass() {
```

Comment:
@ijuma Thank you for the review. Yes, I have tested that they work. Because all the threads that we are looking for rely on ZooKeeper directly or indirectly, and `ZooKeeperTestHarness` uses instance methods to start ZooKeeper, I think this should be safe. I have added a comment.